testing/shuffle.cu

```diff
@@ -1,15 +1,309 @@
 #include <thrust/detail/config.h>
 
 #if THRUST_CPP_DIALECT >= 2011
+#include <map>
+#include <limits>
 #include <thrust/random.h>
 #include <thrust/sequence.h>
 #include <thrust/shuffle.h>
-#include <thrust/sort.h>
 #include <unittest/unittest.h>
-#include <map>
 
-template <typename Vector>
-void TestShuffleSimple() {
+class CephesFunctions {
```

Comment:
Can you add some comments here, explaining what this does and why it's needed? As a maintainer, it's not obvious what this is testing, and the implementation is rather cryptic on its own.